pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -740,6 +741,7 @@ func TestPrometheusCodec_EncodeMetricsQueryRequest_ShouldPropagateHeadersInAllow
 	require.NoError(t, err)
 	require.Equal(t, []string{"true"}, req.Header.Values(compat.ForceFallbackHeaderName))
 	require.Equal(t, []string{"label"}, req.Header.Values(chunkinfologger.ChunkInfoLoggingHeader))
+	require.Equal(t, []string{string(api.EncodeOffsets(map[int32]int64{0: 1, 1: 2}))}, req.Header.Values(api.ReadConsistencyOffsetsHeader))
```

Comment:
This assertion can sometimes flake due to the unpredictable iteration order of maps, for example:

```
--- FAIL: TestPrometheusCodec_EncodeMetricsQueryRequest_ShouldPropagateHeadersInAllowList (0.00s)
    codec_test.go:744:
        	Error Trace:	/mimir/pkg/frontend/querymiddleware/codec_test.go:744
        	Error:      	Not equal:
        	            	expected: []string{"v1=1:2,0:1"}
        	            	actual  : []string{"v1=0:1,1:2"}

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 ([]string) (len=1) {
        	            	- (string) (len=10) "v1=1:2,0:1"
        	            	+ (string) (len=10) "v1=0:1,1:2"
        	            	 }
        	Test:       	TestPrometheusCodec_EncodeMetricsQueryRequest_ShouldPropagateHeadersInAllowList
```

Reply:
Thanks for raising it. I will fix it, thanks.